lib/screens/v2/candidate_generator/dup.ex

```diff
@@ -113,86 +134,212 @@ defmodule Screens.V2.CandidateGenerator.Dup do
-              end
-
-            %{type: :normal_section, rows: visible_departures}
-          end)
-
-        [
-          %DeparturesWidget{
-            screen: config,
-            section_data: sections,
-            slot_names: [:main_content_two]
-          }
-        ]
-      end
+      sections_data_to_departure_instances(
+        config,
+        secondary_sections_data,
+        [:main_content_two],
+        now
+      )
 
     primary_departures_instances ++ secondary_departures_instances
   end
 
+  defp sections_data_to_departure_instances(config, sections_data, slot_ids, now) do
+    if Enum.any?(sections_data, &(&1 == :error)) do
+      %DeparturesNoData{screen: config, show_alternatives?: true}
+    else
+      sections =
+        Enum.map(sections_data, fn %{
+                                     departures: departures,
+                                     pill: pill,
+                                     alert: alert,
+                                     headway: headway,
+                                     stop_ids: stop_ids
+                                   } ->
+          visible_departures =
+            if length(sections_data) > 1 do
+              Enum.take(departures, 2)
+            else
+              Enum.take(departures, 4)
+            end
+
+          case fetch_headway_mode(stop_ids, headway, alert, now) do
+            {:active, time_range, headsign} ->
+              %{type: :headway_section, pill: pill, time_range: time_range, headsign: headsign}
+
+            :inactive ->
+              %{type: :normal_section, rows: visible_departures}
+          end
+        end)
+
+      Enum.map(slot_ids, fn slot_id ->
+        %DeparturesWidget{
+          screen: config,
+          section_data: sections,
+          slot_names: [slot_id]
+        }
+      end)
+    end
+  end
+
+  defp get_sections_data(sections, fetch_section_departures_fn, fetch_alerts_or_empty_list_fn) do
+    Enum.map(sections, fn %Section{
+                            query: %Query{params: %Params{stop_ids: stop_ids} = params},
+                            headway: %Headway{pill: pill} = headway
+                          } = section ->
+      section_alert = get_section_alert(params, fetch_alerts_or_empty_list_fn)
+
+      {:ok, section_departures} = section |> fetch_section_departures_fn.()
+
+      %{
+        departures: section_departures,
+        pill: pill,
+        alert: section_alert,
+        headway: headway,
+        stop_ids: stop_ids
+      }
+    end)
+  end
+
+  defp get_section_alert(
+         %Params{
+           stop_ids: stop_ids,
+           route_ids: route_ids,
+           direction_id: direction_id,
+           route_type: route_type
+         },
+         fetch_alerts_or_empty_list_fn
+       ) do
+    alert_fetch_params = [
+      direction_id: direction_id,
+      route_ids: route_ids,
+      route_types: Util.append_if([:light_rail, :subway], not is_nil(route_type), route_type),
```

Comment:
I did this because we don't always know what `route_type` a section is for. It is something that can be included in the struct, but is not something that is consistently filled out for each screen. Conditionally fetching is possible, but is prone to human error if that part of the config is not populated. I could instead look up the `route_type` through the API using `route_ids`, but that trades one query for another.

Reply:
Totally forgot that `route_type` is in the `departures` data already! Gonna work on that conditional now.

Reply:
Actually, that's not what you meant, and that wouldn't work anyway. The whole point of headway mode is that there are no departures. So I can't get data off of something that doesn't exist 🤦 . My original comment is back in play then. So the options are use config that may not exist to conditionally fetch (minimizes fetches but prone to human error), or hardcode params in fetch so sections for bus, CR, Ferry, SL sections do not return alerts (more fetches, but not a big deal performance wise). What are your thoughts?

Reply:
OH! I didn't realize a configuration might not provide `route_ids`, but might just use `route_type`, or even just `stop_ids`. Ah. Ok. So the config for a section will include some combo of `route_id`, `route_type`, and `stop_ids`, so all three need to be make their way to the alerts filter.

Then my next q about this: why do we need to modify the route_types with the append? Why don't we use identical params for fetching alerts as we do for fetching departures? 

Since we don't ever want to show headway mode for bus or CR, we also need a conditional that only fetches alerts when we're dealing with :light_rail or :subway